lessons/introR/main.Rmd

```diff
@@ -0,0 +1,294 @@
+* It has excellent resources for creating reproducible documents.
+* Its extensive and active community of users for doing statistical
+  work:
+    - R is the top tag in the
+      [StackExchange CrossValidated site](stats.stackexchange.com/tags)
+      for statistical questions
+    - There are more than 6000 packages on
+      [CRAN](https://cran.r-project.org/web/packages/) that stores all
+      R packages (and installing these packages is straight-forward),
+      from the obscure and cutting-edge statistical techniques to
+      plotting to data wrangling.
+
+I should mention a quick caveat.  While R is a general-purpose
+programming language, it works a bit differently from other languages
+such as Python (it was developed by statisticians after all!).  As
+such, programming in R may not be as intuitive, powerful, or easy as
+it may be in Python (though it can be done), especially if you come
+from a computer science background.  If your work involves a lot of
+programming, I would recommend Python as your main tool.  *However*,
+it never hurts to learn more than one language, especially as R is
+great for data analysis and plotting.
+
+Ok, firstly, I've made this session with some assumptions (see the
+[slides.html](./slides.html)) file.  Briefly I'm assuming you want to
+use R for statistical analysis, plotting, and/or writing up reports.
+I'm using R Markdown to show how to write up documents with R code and
+since getting the data into an analyzable form is the hardest part of
+an analysis, I'm using packages specific to that task.
+
+While you can create functions in R, I won't be going over them.  A
+great resource for R functions is
+[this page from Hadley Wickham's 'Advanced R' book](http://adv-r.had.co.nz/Functions.html)
+
+# R Markdown #
+
+An `.Rmd` or [R Markdown](http://rmarkdown.rstudio.com/) file is a
+[markdown](https://en.wikipedia.org/wiki/Markdown) file that contains
+R code chunks that can be processed to output the results of the R
+code into a generated `.md` file.  This is an incredible (and recent)
+strength of using R, as this then allows you to create html, pdf, or
+Word doc files from the `.md` file using the `rmarkdown` package
+(which relies on [pandoc](pandoc.org)).
+
+On the top of each `.Rmd` file is the
+[YAML](https://en.wikipedia.org/wiki/YAML) front matter, which looks
+like:
+
+```
+---
+title: "Introduction to R"
+author: "Luke Johnston"
+date: "July 23, 2015"
+output: 
+  html_document: 
+    highlight: tango
+    number_sections: yes
+    theme: readable
+    toc: yes
+    
+---
+```
+
+Note the starting and ending `---` 'tags'.  This starts the YAML
+block.
+
+Markdown syntax for formatting is used in `.Rmd`.  Check out the
+[R Markdown documentation](http://rmarkdown.rstudio.com/) for a quick
+tutorial on the syntax.
+
+# Import/export your data #
+
+You'll need to import your data into R to analyze it.  I'm assuming
+the data is already cleaned and ready for analysis.  If at any time
```

Comment:
Ah, thanks for catching that. The `swiss` dataset I use below is part of the R internal datasets